prusti-viper/src/encoder/mir/specifications/specs.rs

```diff
@@ -0,0 +1,139 @@
+use log::{debug, trace};
+use prusti_interface::{
+    environment::Environment,
+    specs::typed::{DefSpecificationMap, LoopSpecification, ProcedureSpecification, Refinable},
+};
+use rustc_hash::FxHashMap;
+use rustc_hir::def_id::{DefId, LocalDefId};
+use std::collections::HashMap;
+
+/// Provides access to specifications, handling refinement if needed
+pub(super) struct Specifications {
```

Comment:
Can this state (actually the fields of `SpecMap`) not be fields of the `SpecificationsState` struct directly?

Reply:
I removed one layer of indirection (the `SpecMap`).
I would still like to keep `Specifications`, which is responsible for invoking the refinement.